scipy/spatial/transform/rotation.py

```diff
@@ -1463,6 +1465,88 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, left=None, right=None, return_indices=False):
+        """Reduce this rotation with the provided rotation groups.
+
+        Parameters
+        ----------
+        left : `Rotation` instance, optional
+            Object containing the left rotation(s). Default value (`None`)
+            corresponds to the identity rotation.
+        right : `Rotation` instance, optional
+            Object containing the right rotation(s). Default value (`None`)
+            corresponds to the identity rotation.
+        return_indices : bool
+            Whether to return the best indices
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+        left_best, right_best: integer ndarray
+            Indices of elements from `left` and `right` used for reduction.
+        """
+        if left is None and right is None:
+            return self.__class__(self._quat, normalized=True, copy=True)
+        elif right is None:
+            right = Rotation.identity()
+        elif left is None:
+            left = Rotation.identity()
+
+        # Levi-Civita tensor for triple product computations
+        e = np.zeros((3, 3, 3))
+        e[0, 1, 2] = e[1, 2, 0] = e[2, 0, 1] = 1
+        e[0, 2, 1] = e[2, 1, 0] = e[1, 0, 2] = -1
+
+        """
+        We want to calculate the real components of `q = l * p * r`. It can be
+        shown that:
+        ```
+        qs = ls * ps * rs
+             - ls * dot(pv, rv)
+             - ps * dot(lv, rv)
+             - rs * dot(lv, pv)
+             - dot(cross(lv, pv), rv)
+        ```
+        where `ls` and `lv` denote the scalar and vector components of `l`.
+        """
+
+        def split_rotation(R):
+            q = np.atleast_2d(R.as_quat())
+            return q[:, -1], q[:, :-1]
+
+        p = self
+        ps, pv = split_rotation(p)
+        ls, lv = split_rotation(left)
+        rs, rv = split_rotation(right)
+
+        qs = np.abs(np.einsum('i,j,k', ls, ps, rs) -
+                    np.einsum('i,jx,kx', ls, pv, rv) -
+                    np.einsum('ix,j,kx', lv, ps, rv) -
+                    np.einsum('ix,jx,k', lv, pv, rs) -
+                    np.einsum('xyz,ix,jy,kz', e, lv, pv, rv)
+                    )
+        qs = np.reshape(np.rollaxis(qs, 1), (qs.shape[1], -1))
+
+        # Find best indices from scalar components
+        max_ind = np.argmax(np.reshape(qs, (len(qs), -1)), axis=1)
+        left_best = max_ind // len(right)
+        right_best = max_ind % len(right)
+
+        # Reduce the rotation using the best indices
+        reduced = left[left_best] * p * right[right_best]
+        if self._single:
+            reduced = reduced[0]
```

Comment:
For consistency:

left_best = left_best[0]
right_best = right_best[0]

Reply:
Fixed.  Thanks.